docs/06/01/setup-guide.md

```diff
@@ -0,0 +1,255 @@
+
+```tsx
+import type { TextInputProps } from '@gusto/embedded-react-sdk'
+
+const MyCustomTextInput = ({
+  label,
+  description,
+  errorMessage,
+  isRequired,
+  isDisabled,
+  isInvalid,
+  id,
+  name,
+  value,
+  placeholder,
+  onChange,
+  onBlur,
+  inputRef,
+  shouldVisuallyHideLabel,
+  ...props
+}: TextInputProps) => {
+  // Your custom implementation here
+  return (
+    <div className="my-custom-input-wrapper">
+      <label htmlFor={id || name}>{label}</label>
+      <input
+        type="text"
+        id={id || name}
+        name={name}
+        value={value || ''}
+        onChange={e => onChange && onChange(e.target.value)}
+        // ...other props
+      />
+      {errorMessage && <div className="error">{errorMessage}</div>}
+    </div>
+  )
+}
+```
+
+Make sure your component implementation:
+
+- Handles all required props correctly
+- Maintains accessibility features
+- Follows your design system guidelines
+- Properly passes event handlers
+
+To learn more about how each component should be implemented, you can reference the default implementations in the SDK ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/tree/main/src/components/Common/UI)).
+
+### 2. Create Your Component Adapter Object
+
+Create an object that implements the `ComponentsContextType` interface ([View interface on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts)) with your custom components:
+
+```tsx
+import type { ComponentsContextType } from '@gusto/embedded-react-sdk'
+
+const myCustomComponents: ComponentsContextType = {
+  Alert: props => <MyCustomAlert {...props} />,
+  Badge: props => <MyCustomBadge {...props} />,
+  Button: props => <MyCustomButton {...props} />,
+  ButtonIcon: props => <MyCustomButtonIcon {...props} />,
+  CalendarPreview: props => <MyCustomCalendarPreview {...props} />,
+  Card: props => <MyCustomCard {...props} />,
+  Checkbox: props => <MyCustomCheckbox {...props} />,
+  CheckboxGroup: props => <MyCustomCheckboxGroup {...props} />,
+  ComboBox: props => <MyCustomComboBox {...props} />,
+  DatePicker: props => <MyCustomDatePicker {...props} />,
+  Heading: props => <MyCustomHeading {...props} />,
+  Link: props => <MyCustomLink {...props} />,
+  Menu: props => <MyCustomMenu {...props} />,
+  NumberInput: props => <MyCustomNumberInput {...props} />,
+  OrderedList: props => <MyCustomOrderedList {...props} />,
+  ProgressBar: props => <MyCustomProgressBar {...props} />,
+  Radio: props => <MyCustomRadio {...props} />,
+  RadioGroup: props => <MyCustomRadioGroup {...props} />,
+  Select: props => <MyCustomSelect {...props} />,
+  Switch: props => <MyCustomSwitch {...props} />,
+  Table: props => <MyCustomTable {...props} />,
+  Text: props => <MyCustomText {...props} />,
+  TextInput: props => <MyCustomTextInput {...props} />,
+  UnorderedList: props => <MyCustomUnorderedList {...props} />,
+}
+```
+
+### 3. Use GustoProviderCustomUIAdapter
```

Comment:
I think we need to do a better clarification between GustoProvider and GustoProviderCustomUIAdapter. They can supply custom components with each. With GustoProvider we supply the fallbacks vs. if they want to completely eject they can use GustoProviderCustomUIAdapter